pcdsdevices/lasers/qmini.py

```diff
@@ -83,6 +87,47 @@ class QminiSpectrometer(Device):
     fit_stdev = Cpt(EpicsSignalRO, ':STDEV', kind='config')
     fit_chisq = Cpt(EpicsSignalRO, ':CHISQ', kind='config')
 
+    # Save spectra functions
+    def save_data(self):
+        """
+        Save the wavelength and spectrum PVs to a text file
+        """
+        if len(self.file_dest.get().strip()) == 0:
+            # set a default destination for the file if you're lazy or give it whitespace
+            _file = os.getcwd() + '/' + self.name + time.strftime("_%Y-%m-%d_%H%M%S") + '.txt'
+        else:
+            _file = self.file_dest.get()
+        self.log.info('Saving spectrum to disk...')
+        # Let's format to JSON for the science folk with sinful f-string mangling
+        _settings = ['sensitivity_cal', 'correct_prnu', 'correct_nonlinearity',
+                     'normalize_exposure', 'adjust_offset', 'subtract_dark',
+                     'remove_bad_pixels', 'remove_temp_bad_pixels']
+        _data = {'timestamp': time.strftime("%Y-%m-%d %H:%M:%S"),
+                 'exposure (us)': str(self.exposure.get()),
```

Comment:
I'm not sure I understand why these need to be cast to strings.

Reply:
`json.dumps` was being weird for a second and I think I cast everything to a `str` when sorting out the problem and left that in there. These can probably live as the `int`s that they are

Reply:
Ah, but the wavelength & spectra _must_ be cast to str since `Float32` is not `json` compat, but that's on the consumer to handle turning back into a number.